packaging/version.py

```diff
@@ -38,8 +48,57 @@
     "_Version", ["epoch", "release", "dev", "pre", "post", "local"]
 )
 
+# TODO: Remove TYPE_CHECKING guard after dropping Python 3.7 support.
+if TYPE_CHECKING:
+    from typing import Protocol
+
+    class BaseVersion(Protocol):
```

Comment:
Do we need to type `__init__()` here to make sure implementations don't accidentally cause Liskov violations?